streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorTopologyTest.java

```diff
@@ -723,6 +775,20 @@ public void process(final String key, final String value) {
         return () -> processor;
     }
 
+    private <K, V> ProcessorSupplier<K, V> defineWithStores(final Processor<K, V> processor, final Set<StoreBuilder> stores) {
+        return new ProcessorSupplier<K, V>() {
+            @Override
+            public Processor<K, V> get() {
+                return processor;
```

Comment:
This does not seem to be a good implementation? A supplier is supposed to return a new instance on each `get()` call. It might not be an issue now, but might be error prone in the future. 

Maybe accept an supplier as input instead of a `Processor`